core/src/main/kotlin/org/evomaster/core/search/gene/regex/RegexGene.kt

```diff
@@ -21,10 +20,23 @@ class RegexGene(
         disjunctions.randomize(randomness, forceNewValue, allGenes)
     }
 
+    override fun getValueAsRawString(): String {
+        return disjunctions.getValueAsPrintableString(targetFormat = null)
+    }
+
     override fun getValueAsPrintableString(previousGenes: List<Gene>, mode: String?, targetFormat: OutputFormat?): String {
-        return disjunctions.getValueAsPrintableString(previousGenes, mode, targetFormat)
+        val rawValue = getValueAsRawString()
+        when {
+            (targetFormat == null) -> return "\"$rawValue\""
+            targetFormat.isKotlin() -> return "\"$rawValue\""
+                    .replace("\\", "\\\\")
+                    .replace("$", "\\$")
+            else -> return "\"$rawValue\""
+                    .replace("\\", "\\\\")
+        }
     }
```

Comment:
add a TODO comment: likely these issues with Kotlin (and other languages) escapes would apply to all printed string, and could be moved into a single place

Reply:
Added a //TODO comment for future refactor with StringGene.getValueAsPrintableString()